bip-0325.mediawiki

```diff
@@ -26,36 +26,48 @@ A new type of test network would be more suitable for integration testing by org
 
 A new type of network ("signet"), which takes an additional consensus parameter called the challenge (scriptPubKey). The challenge can be a simple pubkey (P2PKH style), or a k-of-n multisig, or any other script you would want.
 
-The witness commitment of the coinbase transaction is extended to include a secondary commitment (the signature/solution):
-
-    1-4 bytes - Push the following (x + 4) bytes
-    4 bytes - Signet header (0xecc7daa2)
-    x bytes - Solution (sigScript)
-
-Any push operations that do not start with the 4 byte signet header are ignored. Multiple push operations with the 4 byte signet header are ignored except for the first entry.
-
-Any signature operations contained within the challenge use SHA256d(modifiedBlockHash), i.e. the double-SHA256 digest of the following data as the sighash:
-
-{|class="wikitable" style="text-align: center;"
-|-
-!Type
-!Size
-!Name
-|-
-|Int32||4||nVersion
-|-
-|Uint256||32||hashPrevBlock
-|-
-|Uint256||32||modifiedMerkleRoot
-|-
-|Uint32||4||nTime
-|-
-|Uint32||4||nBits
-|}
-
-The <code>modifiedMerkleRoot</code> hash is obtained by generating the merkle root of the block transactions, with the coinbase witness commitment as is, without the signet extension. This means the merkle root of the block is different from the merkle root in the signet commitment. This is needed, because the signature can never be included in the very message (in this case, a block) that is being signed. Apart from the signature, to facilitate block generation (mining), the block nonce value is the only other component of the block that the signet signature does not commit to. When grinding proof of work, the extended nonce cannot be used as it would invalidate the signature. Instead, simply resigning the same (or an updated) block will give a new search space.
-
-A block is considered fully validated if the above commitment is found, and its solution is valid. It is recommended that this verification is done directly before or after the witness commitment verification, as the data required to do both is approximately the same.
+The witness commitment of the coinbase transaction is extended to include a secondary commitment (the signature/solution) of either:
+
+    1-4 bytes - Push the following (4 + x + y) bytes
+    4 bytes - Signet scriptSig header (0xecc7daa2)
+    x bytes - scriptSig
+    y bytes - scriptWitness
```

Comment:
I'm starting to wonder if we even need the scriptSig part at all. It seems that the scriptWitness is a superset of the features we get from scriptSig, and any systems incorporating block sig validation will still need to handle segwit stuff, since that's a possibility now.

Reply:
Could do that, but the scriptWitness would be more data. With scriptSig, you can make the signet challenge be the actual script; but with scriptWitness, the challenge has to be p2wpkh or p2wsh and you have to then include either the pubkey or the script that matches the hash in every block solution.

Reply:
That seems really backwards. It should be possible to put the pubkey/script into the challenge directly, IMO, but we can sort that out later.

Reply:
Just make the witness pubkey/script implicit. The client knows it and can insert it where it belongs.

Reply:
Wallets do this all the time(fill out preimage details), now that the block is a "signable transaction" this is even more the case.

Reply:
Out of curiosity @ajtowns why use the witness commitment for this and not an additional `OP_RETURN` with a different magic header?